sql/driver/driver_test.go

```diff
@@ -45,18 +49,611 @@ func cleanup(s *server.TestServer, db *sql.DB) {
+		"Europe/Paris",
+		"Europe/Podgorica",
+		"Europe/Prague",
+		"Europe/Riga",
+		"Europe/Rome",
+		"Europe/Samara",
+		"Europe/San_Marino",
+		"Europe/Sarajevo",
+		"Europe/Simferopol",
+		"Europe/Skopje",
+		"Europe/Sofia",
+		"Europe/Stockholm",
+		"Europe/Tallinn",
+		"Europe/Tirane",
+		"Europe/Tiraspol",
+		"Europe/Uzhgorod",
+		"Europe/Vaduz",
+		"Europe/Vatican",
+		"Europe/Vienna",
+		"Europe/Vilnius",
+		"Europe/Volgograd",
+		"Europe/Warsaw",
+		"Europe/Zagreb",
+		"Europe/Zaporozhye",
+		"Europe/Zurich",
+		"GB",
+		"GB-Eire",
+		"GMT",
+		"GMT+0",
+		"GMT0",
+		"GMT-0",
+		"Greenwich",
+		"Hongkong",
+		"Iceland",
+		"Indian/Antananarivo",
+		"Indian/Chagos",
+		"Indian/Christmas",
+		"Indian/Cocos",
+		"Indian/Comoro",
+		"Indian/Kerguelen",
+		"Indian/Mahe",
+		"Indian/Maldives",
+		"Indian/Mauritius",
+		"Indian/Mayotte",
+		"Indian/Reunion",
+		"Iran",
+		"Israel",
+		"Jamaica",
+		"Japan",
+		"Kwajalein",
+		"Libya",
+		"Mexico/BajaNorte",
+		"Mexico/BajaSur",
+		"Mexico/General",
+		"Navajo",
+		"NZ",
+		"NZ-CHAT",
+		"Pacific/Apia",
+		"Pacific/Auckland",
+		"Pacific/Chatham",
+		"Pacific/Chuuk",
+		"Pacific/Easter",
+		"Pacific/Efate",
+		"Pacific/Enderbury",
+		"Pacific/Fakaofo",
+		"Pacific/Fiji",
+		"Pacific/Funafuti",
+		"Pacific/Galapagos",
+		"Pacific/Gambier",
+		"Pacific/Guadalcanal",
+		"Pacific/Guam",
+		"Pacific/Honolulu",
+		"Pacific/Johnston",
+		"Pacific/Kiritimati",
+		"Pacific/Kosrae",
+		"Pacific/Kwajalein",
+		"Pacific/Majuro",
+		"Pacific/Marquesas",
+		"Pacific/Midway",
+		"Pacific/Nauru",
+		"Pacific/Niue",
+		"Pacific/Norfolk",
+		"Pacific/Noumea",
+		"Pacific/Pago_Pago",
+		"Pacific/Palau",
+		"Pacific/Pitcairn",
+		"Pacific/Pohnpei",
+		"Pacific/Ponape",
+		"Pacific/Port_Moresby",
+		"Pacific/Rarotonga",
+		"Pacific/Saipan",
+		"Pacific/Samoa",
+		"Pacific/Tahiti",
+		"Pacific/Tarawa",
+		"Pacific/Tongatapu",
+		"Pacific/Truk",
+		"Pacific/Wake",
+		"Pacific/Wallis",
+		"Pacific/Yap",
+		"Poland",
+		"Portugal",
+		"PRC",
+		"ROC",
+		"ROK",
+		"Singapore",
+		"Turkey",
+		"UCT",
+		"Universal",
+		"US/Alaska",
+		"US/Aleutian",
+		"US/Arizona",
+		"US/Central",
+		"US/Eastern",
+		"US/East-Indiana",
+		"US/Hawaii",
+		"US/Indiana-Starke",
+		"US/Michigan",
+		"US/Mountain",
+		"US/Pacific",
+		"US/Samoa",
+		"UTC",
+		"W-SU",
+		"Zulu",
+	}
+
 	s, db := setup(t)
 	defer cleanup(s, db)
 
-	year, month, day := 3015, time.August, 30
-	timeVal := time.Date(year, month, day, 3, 34, 45, 345670000, time.UTC)
-	dateVal := time.Date(year, month, day, 0, 0, 0, 0, time.UTC)
+	offsets := make(map[int]struct{})
+
+	// We don't want to spend forever on this test.
+	for startTime := time.Now(); time.Since(startTime) < 10*time.Second; {
+		locationName := locationNames[rand.Intn(len(locationNames))]
+
+		loc, err := time.LoadLocation(locationName)
+		if err != nil {
+			log.Infof("%s: skipping %s", err, locationName)
+			continue
+		}
+
+		_, offset := time.Time{}.In(loc).Zone()
+		if _, ok := offsets[offset]; !ok {
```

Comment:
Actually, it looks like zoneinfo returns an offset based on longitude for dates before the adoption of a standardized time zone: America/New_York had an offset of 4:56 in the distant past: http://play.golang.org/p/aejEm42w78

This is presumably not what you intended here, which shows that we need some tests with known behavior in a given zone instead of just picking random zones and ensuring that everything round-trips correctly.